AdyenCard/Components/Card/CardViewControllerItemsProvider.swift

```diff
@@ -241,7 +283,30 @@ extension CardViewController {
             )
             return item
         }()
-        
+
+        internal func handleSelection(selectedBrand: CardBrand) {
+            coBadgedCardItem.selectableFormItems.forEach { $0.isSelected = false }
```

Comment:
This also seems like logic belonging to coBadgedCardItem. By passing the new value it can handle the selection/unselection logic under its own area.